### ISSUE TITLE
Proposal: enable "Send secrets to workflows from fork pull requests" feature

### ISSUE THREAD

**Issue by jpmcb** [vmware-tanzu/community-edition#1023]
### TLDR

Enable the "Send secrets to workflows from fork pull requests" feature for workflows originating from forks to enable the use of `GH_ACCESS_TOKEN`until TCE and other Tanzu github repos are public. Then, use the read-only `GITHUB_TOKEN` for workflows from forks.

### Proposal

We want to enable the running of CI / CD from forks to better serve contributors and package authors. This way, they can make PRs from their forks and see any MD or shell linting problems along with build errors. To do this, we enabled the "Run workflows from fork pull requests" feature of GitHub enterprise. But, no secrets are sent to workflows based from forks (for security reasons as [documented here](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories)) and the default read-only `GITHUB_TOKEN` does not have enough permissions to complete workflows and pull from the various tanzu repos. Ex: #1021. But, we can enable the "Send secrets to workflows from fork pull requests" feature to send the higher permission `GH_ACCESS_TOKEN` secret. This _should_ enable the secret access token to be sent and allow those workflows to run appropriately.

### Future plan

So, in the future, to prevent any security risk, we will need to:

- [ ] Make tce and other tanzu github repos public
- [ ] Disable the "Send secrets to workflows from fork pull requests" feature
- [x] ~~Remove the `GH_ACCESS_TOKEN`~~
- [x] Replace workflows using `GH_ACCESS_TOKEN` and the `pull_request` target to use the default `GITHUB_TOKEN`
